service/discovery/azure/network.go

```diff
@@ -247,22 +249,31 @@ func (d *azureNetworkDiscovery) publicIPAddressFromLoadBalancer(lb *network.Load
 	if lb.LoadBalancerPropertiesFormat != nil && lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations != nil {
 		for _, publicIpProperties := range *lb.FrontendIPConfigurations {
 
+			if publicIpProperties.PublicIPAddress == nil {
+				continue
+			}
+
 			publicIpAddressName := getFrontendPublicIPAddressName(*publicIpProperties.PublicIPAddress.ID)
```

Comment:
Since you check if `PublicIPAddress` is nil in L252 and you can assume that the corresponding ID should be filled, it is maybe safe to use the de-referencing pointer ("*") here.
But I guess the `to.String` function from the Azure package should be used everywhere, right? To avoid nil pointer references?